### PR TITLE
GEOS-6905 support WCS 2.0 requests embedded in WPS

### PR DESCRIPTION
**Pull request by jdries**
Added support for embedding WCS 2.0 in WPS, also a basic unit tests which checks that an 1.1 and 2.0.1 xml request can be embedded in a WPS request.

### CODE REVIEW

File: src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/InternalWCSInputProvider.java [hunk 1]
```diff
@@ -45,6 +46,9 @@ protected Object getValueInternal(ProgressListener listener) throws Exception {
             KvpRequestReader reader;
             if (version.equals("1.0.0") || version.equals("1.0")) {
                 reader = (KvpRequestReader) context.getBean("wcs100GetCoverageRequestReader");
+                
+            } else if(version.equals("2.0.1") ){
```
Comment (aaime) [inline src/extension/wps/wps-core/src/main/java/org/geoserver/wps/executor/InternalWCSInputProvider.java]: I'd relax this check a bit and also allow for 2.0.0

**Comment by aaime** [comment]
Looking good, just found a minor issue

**Comment by jdries** [comment]
Ok, I added the WCS version. I also had another minor issue with bad handling of unexpected responses from embedded WFS requests, so I simply added that commit within this pull request, hope that's not a problem.

**Comment by aaime** [comment]
Merged, and broke the build due to a missing test dependency, fix here: https://github.com/geoserver/geoserver/commit/a6c84b272562d9353bd95aa8efc8f255ee1debea

Please remember to check that maven builds fine, it does not use the same dependency rules as Eclipse

**Pull request merged**